boards/Makefile.include.llvm

```diff
@@ -0,0 +1,51 @@
+export GDBPREFIX  ?= $(PREFIX)
+export LLVMPREFIX ?= llvm-
+export CC          = clang
+export CXX         = clang++
+export LINK        = $(CC)
+export AS          = $(LLVMPREFIX)as
+export AR          = $(LLVMPREFIX)ar
+export NM          = $(LLVMPREFIX)nm
+# There is no LLVM linker yet, use GNU binutils.
+#export LINKER      = $(LLVMPREFIX)ld
+# objcopy does not have a clear substitute in LLVM, use GNU binutils
+#export OBJCOPY     = $(LLVMPREFIX)objcopy
+export OBJCOPY     = $(PREFIX)objcopy
+export OBJDUMP     = $(LLVMPREFIX)objdump
+export SIZE        = $(LLVMPREFIX)size
+export DBG         = $(GDBPREFIX)gdb
+# LLVM lacks a binutils strip tool as well...
+#export STRIP      = $(LLVMPREFIX)strip
+
+# Tell clang to cross compile
+export CFLAGS     += -target $(TARGET_TRIPLE)
+export CXXFLAGS   += -target $(TARGET_TRIPLE)
+export LINKFLAGS  += -target $(TARGET_TRIPLE)
+
+# Since Clang is not installed as a separate instance for each crossdev target
+# we need to tell it where to look for platform specific includes (Newlib
+# headers instead of Linux/Glibc headers.)
+# On GCC this is done when building the cross compiler toolchain so we do not
+# actually need to specify the include paths for system includes.
+# Ubuntu gcc-arm-embedded toolchain (https://launchpad.net/gcc-arm-embedded)
+# places newlib headers in several places, but the primary source seem to be
+# /etc/alternatives/gcc-arm-none-eabi-include
+# Gentoo crossdev places newlib headers in /usr/arm-none-eabi/include
+# Ubuntu also seem to put a copy of the newlib headers in the same place as
+# Gentoo crossdev, but we prefer to look at /etc/alternatives first.
+# On OSX, newlib includes are possibly located in
+# /usr/local/opt/arm-none-eabi*/arm-none-eabi/include
+NEWLIB_INCLUDE_PATTERNS ?= \
+  /etc/alternatives/gcc-$(TARGET_TRIPLE)-include \
+  /usr/$(TARGET_TRIPLE)/include \
+  /usr/local/opt/$(TARGET_TRIPLE)*/$(TARGET_TRIPLE)/include \
+  #
+# Use the wildcard Makefile function to search for existing directories matching
+# the patterns above. We use the -isystem gcc/clang argument to add the include
+# directories as system include directories.
+NEWLIB_INCLUDES ?= \
+    $(foreach dir, \
+        $(foreach pat, $(NEWLIB_INCLUDE_PATTERNS), $(wildcard $(pat))), \
+        -isystem $(dir))
+
+export INCLUDES += $(NEWLIB_INCLUDES)
```

Comment:
Is newlib an actual module that can be tested for in USEMODULE?
It is listed under PSEUDOMODULES in Makefile.pseudomodules, but I don't know how that part of the build system works.